pages/querying/best-practices.mdx

```diff
@@ -26,7 +26,232 @@ Indexing best practices in general + index hinting + analyze graph as strategies
+in scenarios where the cost or distance between nodes varies, such as road networks 
+(where edges represent distances or travel times) or communication networks 
+(where edges represent latency or cost).
+
+When the goal is to **optimize** a specific metric along the path (e.g., minimizing 
+cost, maximizing profit), a weighted shortest path algorithm can help find the path 
+that optimizes that metric.
+
+
+#### All shortest paths
+Finding all shortest paths is an expansion of the weighted shortest paths problem. 
+The goal of finding the shortest path is obtaining any minimum sum of weights on 
+the path from one node to the other. However, there could be multiple 
+similar-weighted paths, and this algorithm fetches them all.
+
+The following query searches for all shortest paths with a default weight equal to 1:
+
+```cypher
+MATCH path=(n {id: 0})-[:CloseTo *ALLSHORTEST (r, n | 1)]-(m {id: 15})
+RETURN path;
+```
+
+**When to use all shortest?** <br/>
+Use an all shortest paths algorithm when you need to find all shortest paths between 
+nodes in a graph with weighted edges. This is useful when you need to analyze the 
+network structure or when there may be multiple identical shortest paths between 
+pairs of nodes with minimum cost.
+
+<br/>
+
+<Cards>
+    <Card
+    title="Read more about built-in algorithms"
+    href="/advanced-algorithms/built-in-graph-algorithms"
+    />
+</Cards>
+
+
+### Optimizing graph traversals
+Deep path traversal algorithms play a crucial role in exploring and analyzing 
+graph data. In Memgraph, a powerful graph database management system, optimizing 
+these algorithms can significantly enhance performance when dealing with 
+large-scale graph structures. 
+
+Let's dive into some strategies for optimizing traversal algorithms in Memgraph, 
+along with providing examples using Cypher queries.
+
+For example, let's take a following unoptimized query that traverses through the 
+European transportation network. Using the BFS algorithm, we'll find all of the cities 
+London is connected to with the shortest road path available. In other words, we'll 
+find all shortest paths from the starting node with the `name` property being `London` 
+to all of the nodes in the dataset. 
+
+```cypher
+MATCH path=(:City {name: 'London'})-[*BFS]->(:City)
+RETURN path;
+```
+
+This query returns nodes representing cities London is connected to with the 
+transportation network. 
+
+When running a query and filtering by certain property, **avoid** using subsequent 
+filtering using the `WHERE` clause and rather filter in-line.
+
+For example, the query above could be written in the similar, but inefficient, manner:
+
+```cypher
+MATCH path=(n)-[*BFS]->(m)
+WHERE n:City AND n.name = 'London' AND m:City 
+RETURN path;
+```
+
+By not using the inline filtering, the database first scans all nodes and then additionally
+applies filter using the `WHERE` clause, causing longer scanning time and slower 
+performance.
```

Comment:
Nice!

Reply:
The same query, different query plan if there is an index inside, the query planner will look for all filters (`WHERE`) and replace them with index calls, so it is smart in a sense.